static/skin/index.js

```diff
@@ -1,31 +1,65 @@
+};
+
+function queryUrlBuilder() {
+    let url = `${root}/catalog/search?`;
+    Object.keys(basicConfig).forEach((key, idx) => {
+        url += `${key}=${basicConfig[key]}${idx !== Object.keys(basicConfig).length - 1 ? '&' : ''}`;
+    });
+    return url;
+}
 
 function getInnerHtml(node, query) {
     return node.querySelector(query).innerHTML;
 }
 
-async function loadAndDisplayBooks() {
-  fetch(`${root}/catalog/search`).then(async (resp) => {
-    const data = new window.DOMParser().parseFromString(await resp.text(), 'application/xml');
-    const books = data.querySelectorAll('entry');
-    let bookHtml = '';
-    books.forEach((book) => {
-        const link = book.querySelector('link').getAttribute('href');
-        const title =  getInnerHtml(book, 'title');
-        const description = getInnerHtml(book, 'summary');
+async function loadAndDisplayBooks(append = false) {
+    isFetching = true;
+    fetch(queryUrlBuilder()).then(async (resp) => {
+        const data = new window.DOMParser().parseFromString(await resp.text(), 'application/xml');
+        const books = data.querySelectorAll('entry');
+        let bookHtml = '';
+        books.forEach((book) => {
+            const link = book.querySelector('link').getAttribute('href');
+            const title =  getInnerHtml(book, 'title');
+            const description = getInnerHtml(book, 'summary');
 
-        bookHtml += `<a href='${link}' data-id='${getInnerHtml(book, 'id')}'><div class='book'>
-            <div class='book__background' style="background-image: url('${getInnerHtml(book, 'icon')}');">
-            <div class='book__title' title='${title}'>${title}</div>
-            <div class='book__description' title='${description}'>${description}</div>
-            <div class='book__info'>${getInnerHtml(book, 'articleCount')} articles, ${getInnerHtml(book, 'mediaCount')} medias</div>
-            </div>
-        </div></a>`;
+            bookHtml += `<a href='${link}' data-id='${getInnerHtml(book, 'id')}'><div class='book'>
+                <div class='book__background' style="background-image: url('${getInnerHtml(book, 'icon')}');">
+                <div class='book__title' title='${title}'>${title}</div>
+                <div class='book__description' title='${description}'>${description}</div>
+                <div class='book__info'>${getInnerHtml(book, 'articleCount')} articles, ${getInnerHtml(book, 'mediaCount')} medias</div>
+                </div>
+            </div></a>`;
+        });
+        document.querySelector('.book__list').innerHTML = (append ? document.querySelector('.book__list').innerHTML : '') + bookHtml;
+        isEnd = !books.length;
     });
-    document.querySelector('.book__list').innerHTML = bookHtml;
-  });
+    isFetching = false;
 }
 
+window.addEventListener('resize', async () => {
+    if (isFetching || isEnd) return;
+    viewPortHeight = window.innerHeight;
+    basicConfig['count'] = Math.floor(viewPortHeight/100 + 1) * 3;
```

Comment:
1. Dependence of the `count` on the viewport size must be defined in a single place - introduce a helper function.
2. Shouldn't this dependence include the viewport width as well?